### DISCUSSION TITLE
Self hosted: Storage disks and requests missing

### DISCUSSION THREAD

**Discussion by fhandke-fugro** [biigle/core#817] [Q&A]
Hello,

We are hosting our own instance. 

However I noticed that the "Storage" page is missing for users and admins:
![image](https://github.com/biigle/core/assets/78092148/12e13541-0cfd-4859-90b5-2dc6dc2d7eb9)
Also when creating a volume the "Storage disk" option is missing:
![image](https://github.com/biigle/core/assets/78092148/97a07271-6ac4-4c47-913b-0b7bed4872f0)

Is there some setting in the .env to activate this? I also tried to send a storage request via the API and that returned an error that it is not available for the instance.

Another question of mine was if there is a general page explaining all the setting in .env a bit?

For example: If we configure those AWS settings, can we then use the s3 bucket as a source of images/videos? Or is that only possible via configuring a S3 storage disk?
![image](https://github.com/biigle/core/assets/78092148/a2bf5df6-1330-45fa-9561-ee8db6b35278)

Help is appreciated. Thanks!

**Comment by mzur** [comment]
Hi,


the "Storage" option only appears if you install the [biigle/user-storage](https://github.com/biigle/user-storage) module (for storage requests) and/or the [biigle/user-disks](https://github.com/biigle/user-disks) module (for configurable storage disks such as AWS in BIIGLE). If you only want to allow a fixed set of storage disks (e.g. you have only one S3 bucket for all your data) and your users should not be able to upload their own files through BIIGLE, you don't need these modules for your instance.

You can configure the (fixed) storage disks that are visible to admins or regular users with the `VOLUME_ADMIN_STORAGE_DISKS` and `VOLUME_EDITOR_STORAGE_DISKS` [variables](https://github.com/biigle/biigle/blob/master/build/.env.example#L14-L15). These are comma-separated lists of storage disk names from the [filesystems config](https://github.com/biigle/biigle/blob/master/build/config/filesystems.php).

There is no S3 storage disk in the default configuration but you can add one to the [filesystems config](https://github.com/biigle/biigle/blob/master/build/config/filesystems.php). Here is an example:

```php
's3' => [
    'driver' => 's3',
    'read-only' => true,
    'key' => env('AWS_ACCESS_KEY_ID'),
    'secret' => env('AWS_SECRET_ACCESS_KEY'),
    'region' => env('AWS_DEFAULT_REGION'),
    'bucket' => env('AWS_BUCKET'),
    'endpoint' => env('AWS_ENDPOINT'),
    'use_path_style_endpoint' => env('AWS_USE_PATH_STYLE_ENDPOINT', false),
    'stream_reads' => true,
    'http' => [
        'connect_timeout' => 5,
    ],
    'throw' => true,
],
``` 

As you can see, the config uses the variables from the `.env` file.

**Reply by fhandke-fugro** [reply]
Hi Martin,

thanks for the quick and informative response.

I had a quick look at https://github.com/biigle/core/blob/master/DEVELOPING.md . Is it correct that if I follow the steps under the "Build and run the application" section all the common modules will be already pre installed? Or do I still need to individually install them?

**Reply by mzur** [reply]
If you want to host a production instance, you should start here: https://github.com/biigle/biigle

The file you linked are the instructions to set up a development instance. But yes, the development instance has all modules installed. The production instance not (only the essentials).
